File: backend/src/main/java/com/team05/petmeeting/domain/comment/dto/FeedCommentRes.java
package com.team05.petmeeting.domain.comment.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.team05.petmeeting.domain.comment.entity.FeedComment;

import java.time.LocalDateTime;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record FeedCommentRes(
        Long userId,
        String nickname,
        String profileImageUrl,
        Long commentId,
        String content,
        Long feedId,
        LocalDateTime createdAt
) {
    public static FeedCommentRes from(FeedComment comment) {
        return new FeedCommentRes(
                comment.getUser().getId(),
                comment.getUser().getNickname(),
                comment.getUser().getProfileImageUrl(),
                comment.getId(),
                comment.getContent(),
                comment.getFeed().getId(),
                comment.getCreatedAt()
        );
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/comment/dto/FeedCommentRes.kt
package com.team05.petmeeting.domain.comment.dto

import com.fasterxml.jackson.annotation.JsonInclude
import com.team05.petmeeting.domain.comment.entity.FeedComment
import java.time.LocalDateTime

@JsonInclude(JsonInclude.Include.NON_NULL)
data class FeedCommentRes(
    val userId: Long,
    val nickname: String,
    val profileImageUrl: String?,
    val commentId: Long,
    val content: String,
    val feedId: Long,
    val createdAt: LocalDateTime
) {
    companion object {
        @JvmStatic
        fun from(comment: FeedComment): FeedCommentRes {
            return FeedCommentRes(
                comment.user.id,
                comment.user.nickname,
                comment.user.profileImageUrl,
                comment.id,
                comment.content,
                comment.feed.id,
                comment.createdAt
            )
        }
    }
}
